packages/bff/src/utilities/eserviceDocumentUtils.ts

```diff
@@ -0,0 +1,261 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import crypto from "crypto";
+import { Readable } from "node:stream";
+import { XMLParser } from "fast-xml-parser";
+import { bffApi, catalogApi } from "pagopa-interop-api-clients";
+import { FileManager, WithLogger } from "pagopa-interop-commons";
+import { ApiError, genericError } from "pagopa-interop-models";
+import { P, match } from "ts-pattern";
+import YAML from "yaml";
+import { z } from "zod";
+import { config } from "../config/config.js";
+import {
+  ErrorCodes,
+  interfaceExtractingInfoError,
+  invalidInterfaceContentTypeDetected,
+  invalidInterfaceFileDetected,
+  openapiVersionNotRecognized,
+} from "../model/domain/errors.js";
+import { CatalogProcessClient } from "../providers/clientProvider.js";
+import { BffAppContext } from "../utilities/context.js";
+
+// eslint-disable-next-line max-params
+export async function verifyAndCreateEServiceDocument(
+  catalogProcessClient: CatalogProcessClient,
+  fileManager: FileManager,
+  eService: catalogApi.EService,
+  doc: bffApi.createEServiceDocument_Body,
+  descriptorId: string,
+  documentId: string,
+  ctx: WithLogger<BffAppContext>
+): Promise<void> {
+  const contentType = doc.doc.type;
+  if (!contentType) {
+    throw invalidInterfaceContentTypeDetected(
+      eService.id,
+      "invalid",
+      eService.technology
+    );
+  }
+
+  const serverUrls = await processFile(doc, eService.technology, eService.id);
+  const filePath = await fileManager.storeBytes(
+    {
+      bucket: config.s3Bucket,
+      path: config.eserviceDocumentsPath,
+      resourceId: documentId,
+      name: doc.doc.name,
+      content: Buffer.from(await doc.doc.arrayBuffer()),
+    },
+    ctx.logger
+  );
+
+  const calculateChecksum = async (stream: Readable): Promise<string> =>
+    new Promise((resolve, reject) => {
+      const hash = crypto.createHash("sha256");
+
+      stream.on("data", (data) => {
+        hash.update(data);
+      });
+
+      stream.on("end", () => {
+        resolve(hash.digest("hex"));
+      });
+
+      stream.on("error", (err) => {
+        reject(err);
+      });
+    });
```

Comment:
As suggested before, to reduce the verbosity of this file, it might be clearer to move this function to a separate file, such as `packages/bff/src/utilities/fileUtils.ts`.